server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -145,9 +145,12 @@ private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<
 
     private Quotas getQuotas(MailboxPath mailboxPath) throws MailboxException {
         QuotaRoot quotaRoot = quotaRootResolver.getQuotaRoot(mailboxPath);
-        return new Quotas(ImmutableMap.of(new QuotaId(quotaRoot), new Quotas.Quota(ImmutableMap.of(
-                Type.STORAGE, quotaToValue(quotaManager.getStorageQuota(quotaRoot)),
-                Type.MESSAGE, quotaToValue(quotaManager.getMessageQuota(quotaRoot))))));
+        return Quotas.from(
+            QuotaId.fromQuotaRoot(quotaRoot),
+            Quotas.Quota.from(
```

Comment:
can we get rid of the map ? and What about importing Quotas.Quota ?

Reply:
is it better like this? I introduced generics to be sure to match the good value.
I don't import Quotas.Quota because I already import org.apache.james.mailbox.model.Quota